sde_collections/models/candidate_url.py

```diff
@@ -79,13 +81,99 @@ class CandidateURL(models.Model):
         default=False,
         help_text="Helps keep track if the Current URL is present in production or not",
     )
+    is_tdamm = models.BooleanField("Is TDAMM?", default=False, help_text="Enable TDAMM tagging for this URL")
+
+    TDAMM_TAG_CHOICES = [
+        ("MMA_M_EM", "Messenger - EM Radiation"),
+        ("MMA_M_EM_G", "Messenger - EM Radiation - Gamma rays"),
+        ("MMA_M_EM_X", "Messenger - EM Radiation - X-rays"),
+        ("MMA_M_EM_U", "Messenger - EM Radiation - Ultraviolet"),
+        ("MMA_M_EM_O", "Messenger - EM Radiation - Optical"),
+        ("MMA_M_EM_I", "Messenger - EM Radiation - Infrared"),
+        ("MMA_M_EM_M", "Messenger - EM Radiation - Microwave"),
+        ("MMA_M_EM_R", "Messenger - EM Radiation - Radio"),
+        ("MMA_M_G", "Messenger - Gravitational Waves"),
+        ("MMA_M_G_CBI", "Messenger - Gravitational Waves - Compact Binary Inspiral"),
+        ("MMA_M_G_S", "Messenger - Gravitational Waves - Stochastic"),
+        ("MMA_M_G_CON", "Messenger - Gravitational Waves - Continuous"),
+        ("MMA_M_G_B", "Messenger - Gravitational Waves - Burst"),
+        ("MMA_M_C", "Messenger - Cosmic Rays"),
+        ("MMA_M_N", "Messenger - Neutrinos"),
+        ("MMA_O_BI", "Objects - Binaries"),
+        ("MMA_O_BI_BBH", "Objects - Binaries - Binary Black Holes"),
+        ("MMA_O_BI_BNS", "Objects - Binaries - Binary Neutron Stars"),
+        ("MMA_O_BI_C", "Objects - Binaries - Cataclysmic Variables"),
+        ("MMA_O_BI_N", "Objects - Binaries - Neutron Star-Black Hole"),
+        ("MMA_O_BI_B", "Objects - Binaries - Binary Pulsars"),
+        ("MMA_O_BI_W", "Objects - Binaries - White Dwarf Binaries"),
+        ("MMA_O_BH", "Objects - Black Holes"),
+        ("MMA_O_BH_AGN", "Objects - Black Holes - Active Galactic Nuclei"),
+        ("MMA_O_BH_IM", "Objects - Black Holes - Intermediate mass"),
+        ("MMA_O_BH_STM", "Objects - Black Holes - Stellar mass"),
+        ("MMA_O_BH_SUM", "Objects - Black Holes - Supermassive"),
+        ("MMA_O_E", "Objects - Exoplanets"),
+        ("MMA_O_N", "Objects - Neutron Stars"),
+        ("MMA_O_N_M", "Objects - Neutron Stars - Magnetars"),
+        ("MMA_O_N_P", "Objects - Neutron Stars - Pulsars"),
+        ("MMA_O_N_PWN", "Objects - Neutron Stars - Pulsar Wind Nebula"),
+        ("MMA_O_S", "Objects - Supernova Remnants"),
+        ("MMA_S_F", "Signals - Fast Radio Bursts"),
+        ("MMA_S_G", "Signals - Gamma-ray Bursts"),
+        ("MMA_S_K", "Signals - Kilonovae"),
+        ("MMA_S_N", "Signals - Novae"),
+        ("MMA_S_P", "Signals - Pevatrons"),
+        ("MMA_S_ST", "Signals - Stellar flares"),
+        ("MMA_S_SU", "Signals - Supernovae"),
+    ]
+
+    # Define TDAMM fields but make them optional
+    @property
```

Comment:
See if you are able to find an architecture that allows us to only define a single field and generate two. The idea would be to simplify the definitions and the logic within the models file and the serializers by abstracting it to the DualField generator. 

If you aren't able to figure out a good architecure, then it is ok to implement the more straightforward approach, where we have some duplication in the definitions, but a relatively clean logic contained within the serializers. Something like this:

```
class CandidateURL(models.Model):
    """A candidate URL scraped for a given collection."""

    TDAMM_tag = models.CharField(
    TDAMM_tag_ml = models.CharField(

    division = models.enums()
    division_ml =  

    document_type =
    document_type_ml = 

    ej_classification = 
    ej_classification_ml = 


def serialize()
    tdamm = self.TDAMM_tam if self.TDAMM_tag else self.TDAMM_tag_ml
    division = self.division if self.division else self.division_ml
    document_type = self.document_type if self.document_type else self.division_ml
    ej_classification = self.ej_classification if self.ej_classification else self.division_ml
```

Reply:
Implemented an architecture that allows us to only define a single field with its field_type and generate two.